tests/containers/podman_quadlet.pm

```diff
@@ -67,30 +102,52 @@ _EOF_
     # start the generator and check whether the files are generated
     systemctl("daemon-reload");
     check_unit_states();
-    systemctl("is-active sleeper.service", expect_false => 1);
-    systemctl("is-active sleeper-volume.service", expect_false => 1);
+    for my $unit (@units) {
+        systemctl("is-active $unit", expect_false => 1);
+    }
+
+    # start build unit
+    systemctl("start $unit_name-build.service");
+    record_info('Build output', script_output("journalctl --no-pager -u $unit_name-build"));
+
+    # validate image build
+    systemctl("is-active $unit_name-build.service");
+    validate_script_output('podman images -n', qr/$build_imagetag/);
 
     # start the container
-    systemctl("start sleeper-volume.service");
-    systemctl("start sleeper.service");
+    systemctl("start $unit_name.service");
     check_unit_states();
-    systemctl("is-active sleeper.service");
-    systemctl("is-active sleeper-volume.service");
+    systemctl("is-active $unit_name.service");
+    systemctl("is-active $unit_name-pod.service");
+    systemctl("is-active $unit_name-volume.service");
+    systemctl("is-active $unit_name-network.service");
+
+    # check if the network exists on host
+    my $podman_net = JSON->new->decode(script_output("podman network inspect systemd-$unit_name"));
+    my $net_if = $podman_net->[0]->{network_interface};
+    assert_script_run("ip a show $net_if");
 
     # container checks
-    validate_script_output('podman container list', qr/systemd-sleeper/);
-    validate_script_output('podman volume list', qr/systemd-sleeper/);
+    validate_script_output('podman container list --noheading', qr/systemd-$unit_name/);
+    validate_script_output('podman pod list -n', qr/systemd-$unit_name/);
+    validate_script_output('podman volume list -n', qr/systemd-$unit_name/);
 
     # disable service & remove units
-    systemctl("disable --now sleeper.service");
-    systemctl("disable --now sleeper-volume.service");
-    systemctl("is-active sleeper.service", expect_false => 1);
-    systemctl("is-active sleeper-volume.service", expect_false => 1);
+    systemctl("disable --now $unit_name.service");
+    systemctl("stop $unit_name-volume.service");
+    systemctl("stop $unit_name-network.service");
+    systemctl("stop $unit_name-build.service");
+    for my $unit (@units) {
+        systemctl("is-active $unit", expect_false => 1);
+    }
 }
 
 sub cleanup {
-    script_run("rm -f /etc/containers/systemd/sleeper.container /etc/containers/systemd/sleeper.volume");
+    script_run(sprintf("rm -f %s", join(' ', map { $_->[0] } @files)));
     systemctl("daemon-reload");
+    script_run("podman rmi $build_imagetag");
+    script_run("podman network rm systemd-$unit_name");
+    script_run("podman volume rm systemd-$unit_name");
```

Comment:
We run `podman system prune -f` in `cleanup_system_host()`. I assume it should be enough